src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -0,0 +1,94 @@
+package org.broadinstitute.dsde.rawls.dataaccess.slick
+
+import java.sql.Timestamp
+import java.util.UUID
+import org.joda.time.DateTime
+
+/**
+ * Created by mbemis on 2/18/16.
+ */
+
+case class SubmissionRecord(id: UUID,
+                            submissionDate: Timestamp,
+                            submitterId: String,
+                            methodConfigurationId: Long,
+                            submissionEntityId: Option[Long],
+                            status: Long
+                             )
+
+case class SubmissionStatusRecord(id: Long, name: String)
+
+case class SubmissionWorkflowRecord(submissionId: UUID, workflowId: UUID)
+
+case class SubmissionWorkflowFailureRecord(submissionId: UUID, workflowFailureId: Long)
+
+case class SubmissionValidationRecord(workflowId: UUID, valueId: Option[Long], errorText: Option[String], inputName: String)
+
+trait SubmissionComponent {
+  this: DriverComponent with RawlsUserComponent with MethodConfigurationComponent with EntityComponent with AttributeComponent with WorkflowComponent =>
+
+  import driver.api._
+
+  class SubmissionTable(tag: Tag) extends Table[SubmissionRecord](tag, "SUBMISSION") {
+    def id = column[UUID]("ID", O.PrimaryKey)
+    def submissionDate = column[Timestamp]("DATE_SUBMITTED")
+    def submitterId = column[String]("SUBMITTER")
+    def methodConfigurationId = column[Long]("METHOD_CONFIG_ID")
+    def submissionEntityId = column[Option[Long]]("ENTITY_ID")
+    def statusId = column[Long]("STATUS")
+
+    def * = (id, submissionDate, submitterId, methodConfigurationId, submissionEntityId, statusId) <> (SubmissionRecord.tupled, SubmissionRecord.unapply)
+
+    def submitter = foreignKey("FK_SUB_SUBMITTER", submitterId, rawlsUserQuery)(_.userSubjectId)
+    def methodConfiguration = foreignKey("FK_SUB_METHOD_CONFIG", methodConfigurationId, methodConfigurationQuery)(_.id)
+    def submissionEntity = foreignKey("FK_SUB_ENTITY", submissionEntityId, entityQuery)(_.id)
+    def status = foreignKey("FK_SUB_STATUS", statusId, submissionStatusQuery)(_.id)
+  }
+
+  class SubmissionStatusTable(tag: Tag) extends Table[SubmissionStatusRecord](tag, "SUBMISSION_STATUS") {
+    def id = column[Long]("ID", O.PrimaryKey, O.AutoInc)
+    def name = column[String]("NAME")
+
+    def * = (id, name) <> (SubmissionStatusRecord.tupled, SubmissionStatusRecord.unapply)
```

Comment:
Earlier we had decided that EntityTypes would not have their own table. Do we want to also abandon the tables for SubmissionStatuses?